tests/test_rechunk.py

```diff
@@ -35,6 +37,75 @@ def target_chunks(request):
     return request.param
 
 
+def test_invalid_executor():
+    with pytest.raises(ValueError, match="unrecognized executor"):
+        api._get_executor("unknown")
+
+
+@pytest.mark.parametrize("shape", [(100, 50)])
+@pytest.mark.parametrize("source_chunks", [(10, 50)])
+@pytest.mark.parametrize("target_chunks", [(20, 10)])
+@pytest.mark.parametrize("max_mem", ["10MB"])
+@pytest.mark.parametrize("pass_temp", [True, False])
+@pytest.mark.parametrize("executor", ["dask", api._get_executor("dask")])
```

Comment:
Hm it doesn't, but I can't tell if it should.  I'm converting all the arrays in the dataset to dask and sending them through the dask.array.Array codepath, which is apparently also broken for all other executors.  This is the trace I get in one form or another from different executors:

```
Traceback (most recent call last):
  File "/Users/eczech/repos/pydata/rechunker/tests/test_rechunk.py", line 100, in test_rechunk_dataset
    rechunked.execute()
  File "/Users/eczech/repos/pydata/rechunker/rechunker/api.py", line 77, in execute
    self._executor.execute_plan(self._plan, **kwargs)
  File "/Users/eczech/repos/pydata/rechunker/rechunker/executors/python.py", line 31, in execute_plan
    plan()
  File "/Users/eczech/repos/pydata/rechunker/rechunker/executors/python.py", line 44, in _execute_all
    task()
  File "/Users/eczech/repos/pydata/rechunker/rechunker/executors/python.py", line 39, in _direct_array_copy
    target[key] = source[key]
  File "/Users/eczech/.conda/envs/rechunker-dev/lib/python3.7/site-packages/zarr/core.py", line 1115, in __setitem__
    self.set_basic_selection(selection, value, fields=fields)
  File "/Users/eczech/.conda/envs/rechunker-dev/lib/python3.7/site-packages/zarr/core.py", line 1210, in set_basic_selection
    return self._set_basic_selection_nd(selection, value, fields=fields)
  File "/Users/eczech/.conda/envs/rechunker-dev/lib/python3.7/site-packages/zarr/core.py", line 1501, in _set_basic_selection_nd
    self._set_selection(indexer, value, fields=fields)
  File "/Users/eczech/.conda/envs/rechunker-dev/lib/python3.7/site-packages/zarr/core.py", line 1550, in _set_selection
    self._chunk_setitem(chunk_coords, chunk_selection, chunk_value, fields=fields)
  File "/Users/eczech/.conda/envs/rechunker-dev/lib/python3.7/site-packages/zarr/core.py", line 1665, in _chunk_setitem
    fields=fields)
  File "/Users/eczech/.conda/envs/rechunker-dev/lib/python3.7/site-packages/zarr/core.py", line 1687, in _chunk_setitem_nosync
    chunk = value.astype(self._dtype, order=self._order, copy=False)
  File "/Users/eczech/.conda/envs/rechunker-dev/lib/python3.7/site-packages/dask/array/core.py", line 1843, in astype
    "arguments: {0!s}".format(list(extra))
TypeError: astype does not take the following keyword arguments: ['order']
```

Two potential solutions I see are:

1. Coerce all dataset variables to zarr when not using the dask executor
2. Make the other executors work on dask arrays

The first seems like a bad idea, and I'm not sure how to do the second.

Reply:
I suspect it would just to change `target[key] = source[key]` to `target[key] = np.asarray(source[key])`  inside `_direct_array_copy`? Or perhaps this fix could be done upstream inside `__setitem__` on Zarr arrays?

Anyways, this can definitely be saved for later!